tools/project_manager/show_good_word_suggestions.php

```diff
@@ -37,9 +38,27 @@
 if($format=="update") {
     $postedWords = parse_posted_words($_POST);
 
-    $words = load_project_good_words($projectid);
-    $words = array_merge($words,$postedWords);
-    save_project_good_words($projectid,$words);
+    $is_reject = isset($_POST[REJECT_SUGGESTIONS]);
+    $destination_list = isset($_POST[BAD_WORD_LIST]) ? BAD_WORD_LIST : GOOD_WORD_LIST;
+
+    if ($is_reject)
+    {
+        $postedWords = array();
+    }
+
+    if ($destination_list == GOOD_WORD_LIST)
+    {
+        $words = load_project_good_words($projectid);
+        $words = array_merge($words,$postedWords);
+        save_project_good_words($projectid,$words);
+    }
+    else
+    {
+        var_dump("bad word list");
+        $words = load_project_bad_words($projectid);
+        $words = array_merge($words,$postedWords);
+        save_project_bad_words($projectid,$words);
+    }
```

Comment:
Oh hey - here is a fun bug in the old file that is more evident with your code. We determine the last change for the good words file at the top of the file:
```php
$fileObject = get_project_word_file($projectid,"good");
$timeCutoff = get_integer_param($_REQUEST, 'timeCutoff', $fileObject->mod_time, 0, null);
```
But we save the good words list later, changing when it was last updated. This means that after the page is submitted, it will still show the old suggestions instead of the new ones based on the updated time. This is most evident when we "reject all". If we move the above two lines below this `if($format=="update") {` block that should fix it.

Reply:
I noticed that and was planning on investigating. This makes perfect sense. Updated.